serde-support/src/main/java/io/micronaut/serde/support/serializers/SpecificObjectSerializer.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright 2017-2021 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.serde.support.serializers;
+
+import io.micronaut.core.annotation.Internal;
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.core.beans.exceptions.IntrospectionException;
+import io.micronaut.core.type.Argument;
+import io.micronaut.core.util.CollectionUtils;
+import io.micronaut.serde.Encoder;
+import io.micronaut.serde.Serializer;
+import io.micronaut.serde.exceptions.SerdeException;
+import io.micronaut.serde.reference.PropertyReference;
+import io.micronaut.serde.reference.SerializationReference;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Fallback {@link Serializer} for general {@link Object} values. For deserialization, deserializes to
+ * standard types
+ * like {@link Number}, {@link String}, {@link Boolean}, {@link Map} and {@link List}.
+ * <p>
+ * This class is used in multiple scenarios:
+ * <ul>
+ *     <li>When the user has an {@link Object} property in a serializable bean.</li>
+ *     <li>When the user explicitly calls {@link io.micronaut.json.JsonMapper#writeValue}{@code (gen, }{@link Object}{@code
+ *     .class)}</li>
+ * </ul>
+ */
+@Internal
+public class SpecificObjectSerializer<T> implements Serializer<T> {
```

Comment:
Rename to `CustomizedObjectSerializer`. Can this be made `final`?

Reply:
There is a code that creates a custom class and modifies properties order etc